### PR TITLE
chore: make games dependant on feature

### PR DESCRIPTION
**Pull request by BlockCat**

### CODE REVIEW

**Comment by KarelPeeters** [comment]
Hey, thanks for the PR!

I've been meaning to do something similar for a long time, so thanks for the nudge. Would it be possible to have a separate feature per game instead (eg. `game_chess`, `game_arimaa`, ...) and then a feature that combines all of them (eg. `game_all`)? Then users can ensure they only have minimal dependencies for what they're doing.

**Comment by BlockCat** [comment]
Hoi, goed idea, will do when I have time.

File: src/ai/mcts.rs [hunk 1]
```diff
@@ -364,8 +364,7 @@ pub fn mcts_build_tree<B: AltBoard>(
 
     let mut tree = Tree::new(root_board.clone());
 
-    let root_outcome = root_board.outcome().map(|o| o.pov(root_board.next_player().other()));
-    tree.nodes.push(Node::new(None, root_outcome));
+    tree.nodes.push(Node::new(None, None));
```
Comment (KarelPeeters) [inline src/ai/mcts.rs]: Why this change? I think this introduces a bug!

File: src/interface/uai/mod.rs [hunk 2]
```diff
@@ -3,5 +3,6 @@
 //! A derivative of the UCI protocol for the game Ataxx.
 //! Loose specification available at <https://ataxx.org/#comm>.
 
+#[cfg(feature = "game_ataxx")]
```
Comment (KarelPeeters) [inline src/interface/uai/mod.rs]: I think this is redundant, the entire `uai` module is already feature-gated.

**Review by KarelPeeters — CHANGES_REQUESTED** [review]
Looks good! If you make the last couple of changes I'll test everything locally and then merge the PR.

File: src/ai/mcts.rs (hunk 1 above)
Comment (BlockCat) [reply src/ai/mcts.rs]: Oops, this was not supposed to go with it.
I changed it to this because `root_board` is not done, and then the outcome will always be None

Comment (KarelPeeters) [reply src/ai/mcts.rs]: Oh you're right, there's an assert right above that checks that the board is not done.

Ideally the all the MCTS stuff would keep working even if the root board is done, and just return `None` in all the right places, but that's indeed a separate issue.

**Comment by KarelPeeters** [comment]
Thanks for the PR! I've merged it and made a couple of small followup changes (documentation, a small bug in a test, adding a check to CI).
